homeassistant/components/history_stats/data.py

```diff
@@ -223,3 +233,18 @@ def _async_compute_seconds_and_changes(
         # Save value in seconds
         seconds_matched = elapsed
         return seconds_matched, match_count
+
+    def _prune_history_cache(self, start_timestamp: float) -> None:
+        """Remove unnecessary old data from the history state cache from previous runs.
+
+        Update the timestamp of the last record from before the start to the current start time.
+        """
+        trim_count = 0
+        for i, history_state in enumerate(self._history_current_period):
+            if history_state.last_changed >= start_timestamp:
+                break
+            history_state.last_changed = start_timestamp
```

Comment:
I looked at this again, I think we could move this line into the if block, since there's only one of these, we actually need to update.